### PR TITLE
Restyle Modals

### PR DESCRIPTION
**Pull request by meganniu**
Summary:
+ styling in `modal.scss` was organized into separate files (mainly `formModal.scss` and `logoModal.scss`)
+ `FormModal` wraps a form
+ Delete Request Type dialog has been separated into its own component 

### Links
STAGING: https://bp-pregnancy-centre-staging.web.app/

[Figma](https://www.figma.com/file/mhy5U4cN5pGjygnVEvZmpg/MVP%2B1-Features?node-id=2%3A46)

#### Request Form
Screenshot from 2021-06-07 20-44-47
Screenshot from 2021-06-07 20-45-09

#### Request Type Form
Screenshot from 2021-06-07 20-45-51
Screenshot from 2021-06-07 20-45-56
Screenshot from 2021-06-07 20-46-07

#### Request Group Form
Screenshot from 2021-06-07 20-47-03
Screenshot from 2021-06-07 20-47-21

#### Authentication Modals
Screenshot from 2021-06-07 20-43-58
Screenshot from 2021-06-07 20-43-40

#### Dropdown
Screenshot from 2021-06-07 21-47-31

### CODE REVIEW

**Comment by judyzhong27** [comment]
The quantity form field should not be that stretched out - it's 106px long

New Request Modal
- Order of form fields should be Full Name > Need > Type > Quantity
- Filler text for the name is no longer "ex. Julia Michaels" it's "Enter the client’s full name"
- When a form field is in error state, the form field outline AND the filler text should be in red

Create Type Modal
- Form field filler text turns red in error state E7B0B5
- Let's remove the title altogether
- Delete type looks good

Create New Need modal
- "Group Name" should just be "Name"
- There is no error state for types. They don't have to create "at least 1". If they don't create anything, they can leave it blank and a "One Size" type will be created by default
- There is no exclamation icon in this modal for error states. See design below
Screen Shot 2021-06-07 at 9 53 49 PM
- Filler text for form should be red
- Since they can upload and select an image now, the language is "Add an Image" not "Select an Image"

Login modal
- Space between Email Address and Sign in to continue should be 40 px. It looks way too large right now
- The default form field's outline is lighter than the one above. Both should be 797979 outline

Dropdown
- When the dropdown clicked, that also means the form field is clicked. Therefore, it should be in a typing state. See screenshot
Screen Shot 2021-06-07 at 9 58 58 PM

**Comment by meganniu** [comment]
Thanks for the review @judyzhong27!

A few clarifications:
- this PR **does not** include the mods for a default request type (that will be a differently PR in a few weeks probably)
- Image upload will be added in a different PR. the copy on the request group modal will change in that PR

New Request Modal
- Order of form fields should be Full Name > Need > Type > Quantity ✅
- Filler text for the name is no longer "ex. Julia Michaels" it's "Enter the client’s full name" ✅
- When a form field is in error state, the form field outline AND the filler text should be in red ✅

Screenshot from 2021-06-12 15-17-59 

Create Type Modal
- Form field filler text turns red in error state E7B0B5 ✅
- Let's remove the title altogether ❓

Screenshot from 2021-06-12 15-40-31


Create New Need modal
- "Group Name" should just be "Name" ✅
- There is no exclamation icon in this modal for error states. ✅
- Filler text for form should be red ✅

Screenshot from 2021-06-12 16-00-23

### Login modal
- Space between Email Address and Sign in to continue should be 40 px. It looks way too large right now ✅
- The default form field's outline is lighter than the one above. Both should be 797979 outline ✅

Screenshot from 2021-06-12 17-02-57

### Dropdown
- When the dropdown clicked, that also means the form field is clicked. Therefore, it should be in a typing state. ✅

https://user-images.githubusercontent.com/20423590/121786482-03887080-cb8e-11eb-877f-bff97451bf1d.mov

File: client/src/components/atoms/Button.tsx [hunk 1]
```diff
@@ -3,15 +3,17 @@ import React, { FunctionComponent } from "react";
 import {CopyToClipboard} from 'react-copy-to-clipboard';
 
 interface ButtonProps {
+    className?: string,
     text: string,
     onClick?: React.MouseEventHandler<HTMLButtonElement>,
-    copyText: string
+    copyText: string,
+    type?: "button" | "submit" | "reset" | undefined,
```
Comment (kevin-wangg) [inline client/src/components/atoms/Button.tsx:10]: Just curious, are these going to be the only types of buttons we'll have?

**Review by kevin-wangg — APPROVED** [review]
LGTM!

File: client/src/components/atoms/Button.tsx (hunk 1 above)
Comment (meganniu) [reply client/src/components/atoms/Button.tsx:10]: Yup! they're taken from the type attribute on html buttons (see [here](https://www.w3schools.com/tags/att_button_type.asp))

**Review by guy-ro — APPROVED** [review]
LGTM

**Review by Anand-Issac — APPROVED** [review]
LGTM

**Pull request merged**